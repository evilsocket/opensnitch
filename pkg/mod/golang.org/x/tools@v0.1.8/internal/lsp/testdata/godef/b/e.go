package b

import (
	"fmt"

	"golang.org/x/tools/internal/lsp/godef/a"
)

func useThings() {
	t := a.Thing{}      //@mark(bStructType, "ing")
	fmt.Print(t.Member) //@mark(bMember, "ember")
	fmt.Print(a.Other)  //@mark(bVar, "ther")
	a.Things()          //@mark(bFunc, "ings")
}

/*@
godef(bStructType, Thing)
godef(bMember, Member)
godef(bVar, Other)
godef(bFunc, Things)
*/

func _() {
	var x interface{}      //@mark(eInterface, "interface{}")
	switch x := x.(type) { //@hoverdef("x", eInterface)
	case string: //@mark(eString, "string")
		fmt.Println(x) //@hoverdef("x", eString)
	case int: //@mark(eInt, "int")
		fmt.Println(x) //@hoverdef("x", eInt)
	}
}
