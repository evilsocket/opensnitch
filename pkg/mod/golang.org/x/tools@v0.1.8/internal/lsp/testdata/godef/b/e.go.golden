-- Member-definition --
godef/a/d.go:6:2-8: defined here as ```go
field Member string
```

[`(a.Thing).Member` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Thing.Member)

\@Member
-- Member-definition-json --
{
	"span": {
		"uri": "file://godef/a/d.go",
		"start": {
			"line": 6,
			"column": 2,
			"offset": 90
		},
		"end": {
			"line": 6,
			"column": 8,
			"offset": 96
		}
	},
	"description": "```go\nfield Member string\n```\n\n[`(a.Thing).Member` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Thing.Member)\n\n\\@Member"
}

-- Member-hoverdef --
```go
field Member string
```

[`(a.Thing).Member` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Thing.Member)

\@Member
-- Other-definition --
godef/a/d.go:9:5-10: defined here as ```go
var a.Other a.Thing
```

[`a.Other` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Other)

\@Other
-- Other-definition-json --
{
	"span": {
		"uri": "file://godef/a/d.go",
		"start": {
			"line": 9,
			"column": 5,
			"offset": 121
		},
		"end": {
			"line": 9,
			"column": 10,
			"offset": 126
		}
	},
	"description": "```go\nvar a.Other a.Thing\n```\n\n[`a.Other` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Other)\n\n\\@Other"
}

-- Other-hoverdef --
```go
var a.Other a.Thing
```

[`a.Other` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Other)

\@Other
-- Thing-definition --
godef/a/d.go:5:6-11: defined here as ```go
type Thing struct {
	Member string //@Member
}
```

[`a.Thing` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Thing)
-- Thing-definition-json --
{
	"span": {
		"uri": "file://godef/a/d.go",
		"start": {
			"line": 5,
			"column": 6,
			"offset": 65
		},
		"end": {
			"line": 5,
			"column": 11,
			"offset": 70
		}
	},
	"description": "```go\ntype Thing struct {\n\tMember string //@Member\n}\n```\n\n[`a.Thing` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Thing)"
}

-- Thing-hoverdef --
```go
type Thing struct {
	Member string //@Member
}
```

[`a.Thing` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Thing)
-- Things-definition --
godef/a/d.go:11:6-12: defined here as ```go
func a.Things(val []string) []a.Thing
```

[`a.Things` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Things)
-- Things-definition-json --
{
	"span": {
		"uri": "file://godef/a/d.go",
		"start": {
			"line": 11,
			"column": 6,
			"offset": 148
		},
		"end": {
			"line": 11,
			"column": 12,
			"offset": 154
		}
	},
	"description": "```go\nfunc a.Things(val []string) []a.Thing\n```\n\n[`a.Things` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Things)"
}

-- Things-hoverdef --
```go
func a.Things(val []string) []a.Thing
```

[`a.Things` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/a?utm_source=gopls#Things)
-- eInt-hoverdef --
```go
var x int
```
-- eInterface-hoverdef --
```go
var x interface{}
```
-- eString-hoverdef --
```go
var x string
```
